src/config_format/flb_cf_yaml.c

```diff
@@ -109,54 +111,131 @@ enum state {
     STATE_STOP            /* end state */
 };
 
+static char *state_names[] = {
+    "start",           /* start state */
+    "stream",          /* start/end stream */
+    "document",        /* start/end document */
+
+    "section",         /* top level */
+    "section-key",
+    "section-value",
+
+    "service",         /* 'service' section */
+    "include",         /* 'includes' section */
+    "other",           /* any other unknown section */
+
+    "custom",          /* custom plugins */
+    "pipeline",        /* pipeline groups customs inputs, filters and outputs */
+
+    "input",    /* input plugins section */
+    "filter",   /* filter plugins section */
+    "output",   /* output plugins section */
+
+    "plugin-start",
+    "plugin-key",
+    "plugin-value",
+    "plugin-values",
+
+    "group-key",
+    "group-value",
+
+    "processors",
+    "processor",
+    "processor-map",
+
+    /* environment variables */
+    "env",
+
+
+    "stop"            /* end state */
+};
+
+static char *state_str(enum state val)
+{
+    if (val < 0 || val > STATE_STOP) {
+	return "unknown";
+    }
+
+    return state_names[val];
```

Comment:
This implementation can cause bug.
Since we need to add/remove new value to/from `state_names` and `enum state` and they should be same index.
If these indices are misaligned, state_str will return wrong string for multiple `enum state`.

Reply:
I reimplemented it with a switch/case using a return on the same line. That resolves the possible bug from unaligned names and also keeps the code within the context of a single page.